packages/manager/src/managers/prismicRepository/PrismicRepositoryManager.ts

```diff
@@ -0,0 +1,266 @@
+	//
+	// Reply from Angelo 2022-12-22: I think it should be in manager
+	// because we shouldn't be exporting root-level utilities from this
+	// package. If we want to make it more inline with the other methods,
+	// we could simplify the API by changing its signature to the
+	// following:
+	//
+	// ```ts
+	// (repositoryName: string) => Promise<boolean>
+	// ```
+	//
+	// This method would:
+	//
+	// 1. Fetch the list of repositories for the user using `readAll()`.
+	//    The list would be cached.
+	// 2. Determine if the user has write access to the given repository.
+	//
+	// This version has the following benefits:
+	//
+	// - Does not expect the consumer to supply a repository object; it
+	//   only requires a repository name, which could be sourced from
+	//   anything (incl. the project's `sm.json`).
+	//
+	// - Similarly, it does not expect the consumer to call `readAll()`
+	//   before calling this method.
+	//
+	// - Works for repositories that the user does not have access to. For
+	//   example, I could use it to check if I have access to "qwerty",
+	//   even if I am not added as a user. The purpose of the method is
+	//   still valid: do I have write access to a given repository?
+	hasWriteAccess(repository: PrismicRepository): boolean {
+		switch (repository.role) {
+			case PrismicRepositoryRole.SuperUser:
+			case PrismicRepositoryRole.Owner:
+			case PrismicRepositoryRole.Administrator:
+				return true;
+
+			default:
+				return false;
+		}
+	}
+
+	async checkExists(
+		args: PrismicRepositoryManagerCheckExistsArgs,
+	): Promise<boolean> {
+		const url = new URL(
+			`./app/dashboard/repositories/${args.domain}/exists`,
+			API_ENDPOINTS.PrismicWroom,
+		);
+		const res = await this._fetch({ url });
+		const text = await res.text();
+
+		if (res.ok) {
+			return text === "false"; // Endpoint returns `false` when repository exists
+		} else {
+			throw new Error(
+				`Failed to check repository existence for domain \`${args.domain}\``,
+				{ cause: text },
+			);
+		}
+	}
+
+	async create(args: PrismicRepositoryManagerCreateArgs): Promise<void> {
+		const url = new URL(
+			"./authentication/newrepository?app=slicemachine",
+			API_ENDPOINTS.PrismicWroom,
+		);
+
+		const body = {
+			...DEFAULT_REPOSITORY_SETTINGS,
+			domain: args.domain,
+			framework: args.framework, // This property appears to be optional for the API
+		};
+
+		const res = await this._fetch({
+			url,
+			method: "POST",
+			body,
+			userAgent: PrismicRepositoryUserAgent.SliceMachine, // Custom User Agent is required
+		});
+		const text = await res.text();
+
+		// Endpoint returns repository name on success, which must be more than 4 characters and less than 30
+		// if (!res.ok) {
+		if (!res.ok || text.length < 4 || text.length > 30) {
+			throw new Error(`Failed to create repository \`${args.domain}\``, {
+				cause: res,
+			});
+		}
+	}
+
+	// TODO: Delete this endpoint? It doesn't seem to be used (I might be wrong). - Angelo
```

Comment:
If we get confirmation from the team, we could remove this method.

If we decide to abstract this out into a dedicated client (similar to `@prismicio/custom-types-client`), we should probably include the `delete()` method so that it is feature-complete.